pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -104,6 +105,26 @@ type VSphereConfig struct {
 	}
 }
 
+type Volumes interface {
+	// AttachDisk attaches given disk to given node. Current node
+	// is used when nodeName is empty string.
+	AttachDisk(vmDiskPath string, nodeName string) (diskID string, diskUUID string, err error)
+
+	// DetachDisk detaches given disk to given node. Current node
+	// is used when nodeName is empty string.
+	DetachDisk(volPath string, nodeName string) error
```

Comment:
Please modify `DetachDisk()` so that it does *not* fail if the specified `nodeName` does not exist. e.g. PR https://github.com/kubernetes/kubernetes/pull/29485